### ISSUE TITLE
GEOQA_R1V_Train_8K数据集上，grpo训练效果不如sft

### ISSUE THREAD

**Issue by Alyroy** [Deep-Agent/R1-V#148]
使用Qwen2-VL-2B-Instruct 模型 ，grpo训练准确率达31.83，而sft训练的准确率达37.79
这是sft的训练配置
# Model arguments
model_name_or_path: /mnt/pfs-gv8sxa/nlu/security/jjt/code2/R1-V/model/Qwen/Qwen2-VL-2B-Instruct
model_revision: main
torch_dtype: bfloat16

# Data training arguments
dataset_name: /mnt/pfs-gv8sxa/nlu/security/jjt/code2/R1-V/leonardPKU/GEOQA_R1V_Train_8K
dataset_configs:
- all
preprocessing_num_workers: 8

# SFT trainer config
bf16: true
do_eval: true
eval_strategy: "no"
gradient_accumulation_steps: 2
gradient_checkpointing: true
gradient_checkpointing_kwargs:
  use_reentrant: false
#hub_model_id: Qwen2-VL-2B-Instruct-SFT
#hub_strategy: every_save
learning_rate: 2.0e-05
log_level: info
logging_steps: 5
logging_strategy: steps
lr_scheduler_type: cosine
packing: true
max_seq_length: 4096
max_steps: -1
num_train_epochs: 2
output_dir: /mnt/pfs-gv8sxa/nlu/security/jjt/code2/R1-V/save/qwen2vl2b_instruct_geoqa_sft1
overwrite_output_dir: true
per_device_eval_batch_size: 4
per_device_train_batch_size: 1
push_to_hub: false
#report_to:
#- wandb
save_strategy: "no"
seed: 42
warmup_ratio: 0.1

**Comment by Alyroy** [comment]
cd src/r1-v

export DEBUG_MODE="true"
export LOG_PATH="./debug_geqa.txt"



torchrun --nproc_per_node="8" \
    --nnodes="1" \
    --node_rank="0" \
    --master_addr="127.0.0.1" \
    --master_port="12345" \
    src/open_r1/grpo.py \
    --output_dir /mnt/pfs-gv8sxa/nlu/security/jjt/code2/R1-V/save/qwen2B_geoqa \
    --model_name_or_path /mnt/pfs-gv8sxa/nlu/security/jjt/code2/R1-V/model/Qwen/Qwen2-VL-2B-Instruct \
    --dataset_name /mnt/pfs-gv8sxa/nlu/security/jjt/code2/R1-V/leonardPKU/GEOQA_R1V_Train_8K \
    --max_prompt_length 1024 \
    --per_device_train_batch_size 1 \
    --gradient_accumulation_steps 2 \
    --logging_steps 1 \
    --bf16 \
    --report_to wandb \
    --gradient_checkpointing false \
    --attn_implementation flash_attention_2 \
    --max_pixels 401408 \
    --num_train_epochs 2 \
    --run_name  Qwen2-VL-2B-GEOQA_8K \
    --save_steps 100 \
    --save_only_model true \
    --deepspeed local_scripts/zero3.json \  这是grpo训练配置

**Comment by chenllliang** [comment]
It's true, we also discuss it in the blog https://deepagent.notion.site/rlvr-in-vlms . The main reason is the test set of GEOQA is completely in domain with the training set, where we think SFT has more advantage.

**Comment by Alyroy** [comment]
thanks，I get the point with the blog.


However, I still have a question. Why does it take an extremely long time, more than 40 hours, to train GRPO using a single machine with eight A800 GPUs? I'm using the Qwen2VL 2B-instruct model, and this is the case for both the counting task and the geometric reasoning task.




---原始邮件---
发件人: "Liang ***@***.***&gt;
发送时间: 2025年3月3日(周一) 下午2:58
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [Deep-Agent/R1-V] GEOQA_R1V_Train_8K数据集上，grpo训练效果不如sft (Issue #148)


   
It's true, we also discuss it in the blog https://deepagent.notion.site/rlvr-in-vlms . The main reason is the test set of GEOQA is completely in domain with the training set, where we think SFT has more advantage.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;
  chenllliang left a comment (Deep-Agent/R1-V#148)
 
It's true, we also discuss it in the blog https://deepagent.notion.site/rlvr-in-vlms . The main reason is the test set of GEOQA is completely in domain with the training set, where we think SFT has more advantage.

**Comment by sleepyshep** [comment]
Hello! I'm encountering an unexpected discrepancy in my training results. When I trained Qwen2.5-VL-3B-Instruct separately using GRPO and SFT methods, I observed that SFT's performance was significantly inferior to GRPO, which contradicts your findings, as shown in the attached image.

![Image](https://github.com/user-attachments/assets/b64df66e-6eed-40f6-89ee-5572394fbc86)

Could you please share the dataset and configuration you used for SFT?  

Here are my training details using LLaMA-Factory for full fine-tuning of Qwen2.5-VL-3B-Instruct:  

**Dataset:**  
• Source: https://huggingface.co/datasets/leonardPKU/GEOQA_R1V_Train_8K  
• Format:  
```json  
{  
  "messages": [  
    {"content": "<image>In the given diagram, if angle 1 has a measure of 35.0 degrees, what is the measure of angle 2?", "role": "user"},  
    {"content": "145°", "role": "assistant"}  
  ],  
  "images": ["1000.png"]  
}  
```  

**Training Configuration:**  
```yaml  
# Model  
model_name_or_path: Qwen2.5-VL-3B-Instruct  
image_max_pixels: 262144  
video_max_pixels: 16384  
trust_remote_code: true  

# Method  
stage: sft  
do_train: true  
finetuning_type: full  
freeze_vision_tower: true  
freeze_multi_modal_projector: true  
train_mm_proj_only: false  
deepspeed: examples/deepspeed/ds_z2_config.json  

# Dataset  
dataset: GEOQA_train  
template: qwen2_vl  
cutoff_len: 2048  
max_samples: 1000000  
overwrite_cache: true  
preprocessing_num_workers: 16  

# Output  
output_dir: qwen2_5_vl-3b/full/GEOQA_train  
logging_steps: 10  
save_steps: 100  
plot_loss: true  
overwrite_output_dir: true  

# Training Parameters  
per_device_train_batch_size: 1  
gradient_accumulation_steps: 2  
learning_rate: 2.0e-5  
num_train_epochs: 0.3  
lr_scheduler_type: cosine  
warmup_ratio: 0.1  
bf16: true  
ddp_timeout: 180000000  
```  

I’d appreciate any insights or suggestions you might have regarding this performance discrepancy.
